src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -173,23 +173,23 @@ public void LookForModuleLevelAndGlobalSuppressions (ModuleDefinition module, IC
 				if (!TryDecodeSuppressMessageAttributeData (instance, out info))
 					continue;
 
-				if (info.Target == null && (info.Scope == "module" || info.Scope == null)) {
+				// If the scope is missing we treat the suppression as if it was placed on the module.
+				var scope = info.Scope?.ToLower ();
+				if (scope == "module" || scope == null) {
 					AddSuppression (info, provider);
```

Comment:
This looks like it will make it easy to accidentally suppress all warnings in a module if you specify a target but forget the scope - we should add a test for this.

Reply:
Yup, it will follow the same path as TargettedModuleSuppressionWithUnmatchedScope.cs and suppress all warnings in the module. I'll add another test to make this explicit in the case where a scope isn't passed.

Reply:
I think it would be worth making this a warning instead. If I add:
```csharp
[module: UnconditionalSuppressMessage ("Test", "IL2026", Target ="M:MyLib.Program.MyDangerousMethod")]
```
That looks like it suppresses warnings only on `MyDangerousMethod`. Because it does suppress the warning from that method, it's all too easy to assume that it works as expected and leave it as-is, hiding future warnings if the author introduces new RUC code.

Before this change, the suppression would at least be ignored. If you'd rather scope the warning behavior out of this change, we should continue ignoring such suppressions, and track adding a warning for it (I think it's not covered by the existing issues, but I might have missed it).

Reply:
> Because it does suppress the warning from that method, it's all too easy to assume that it works as expected and leave it as-is, hiding future warnings if the author introduces new RUC code.
>
> Before this change, the suppression would at least be ignored.

Ah, yeah. Reverted -- we'll keep ignoring these scenarios.